cordapp/src/main/kotlin/net/corda/plugins/CordappPlugin.kt

```diff
@@ -76,15 +79,57 @@ class CordappPlugin : Plugin<Project> {
         }
         jarTask.dependsOn(task)
 
-        val signTask = project.task("signCordappJar")
-        signTask.doLast {
-            if (cordapp.signing.enabled) {
-                val options = cordapp.signing.options.toSignJarOptionsMap()
-                options["jar"] = project.tasks.getByName("jar").outputs.files.singleFile.toPath().toString()
+        val signTask = project.task("signCordapp") { it.doLast{ sign(project) } }
+        jarTask.finalizedBy(signTask)
+
+        val sourceJarTask = project.tasks.findByName("sourceJar") //task in corda-publish-utils plugin
```

Comment:
Hmm, why is this specific to making source JARs? I am not sure where source JARs enter the picture.

Reply:
Good point. This should be checking for the maven-publish module and it's artifacts. The publish-utils plugin is really only used by R3 and Corda. Cordapp developers will have their own publishing pipeline. 

It'd make more sense to either a) not sign the sources JAR, if that's even an option, or b) search for all JAR tasks in this project and sign them all or c) sign all by default but allow configuration of which tasks get signed

Reply:
The problem here is that for jar task run for development use the Corda dev key is ok (out-of-box) but once it's published at least we need to warn that dev key is in use - so I need a way to distinguish the context in which jar task is run. The jar task  itself doesn't know if it's running just as a part of dev build or publishing task (I could presumably walk the task graph and check if publishing is present).  
The simplification of this process would be as you said `c) sign all by default but allow configuration of which tasks get signed`  but if users forgets to add the specific configuration for own keyStore, then it will be silently published with our dev key

Reply:
Discussed with Clinton, it will be simplified (sign by default but opt-out, info when signing with dev key)  and adding logic to publish-utils from cordapp plugin.